instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -370,7 +370,19 @@ def get_host_port_url_tuple(scope):
     server = scope.get("server") or ["0.0.0.0", 80]
     port = server[1]
     server_host = server[0] + (":" + str(port) if str(port) != "80" else "")
-    full_path = scope.get("root_path", "") + scope.get("path", "")
+    # To get the correct virtual url path within the hosting application (e.g also in a subapplication scenario)
+    # we have to remove the root_path from the path
```

Comment:
This is more an explanation of the change than an explanation of the code. I'd rewrite this last line as "using the scope path is enough".